### ISSUE TITLE
Image Optimization using the default loader is not compatible with export

### ISSUE THREAD

**Issue by haydnjonesdesign** [dc7290/next-export-optimize-images#971]
**Describe the bug**
Next build command fails

```
> next build && next-export-optimize-images

info - [next-export-optimize-images]: Configuration was not loaded. (This is optional.)
  ▲ Next.js 14.2.4

 ✓ Linting and checking validity of types    
   Creating an optimized production build ...
 ✓ Compiled successfully
 ✓ Collecting page data    
 ✓ Generating static pages (4/4)
 ✓ Collecting build traces    

> Image Optimization using the default loader is not compatible with export.
  Possible solutions:
    - Use `next start` to run a server, which includes the Image Optimization API.
    - Configure `images.unoptimized = true` in `next.config.js` to disable the Image Optimization API.
  Read more: https://nextjs.org/docs/messages/export-image-api
   Finalizing page optimization  . ELIFECYCLE  Command failed with exit code 1.
```

**To Reproduce**
Copy and paste steps in Documentation / Getting Started

**Expected behavior**
Build passed next build and continues on to next-export-optimize-images successfully.

 - Node 20.11.1
 - "next": "14.2.4",
 - "next-export-optimize-images": "^4.5.0",

**Additional context**
I've attempted to do this with a simple image on a single page with absolutely nothing fancy. 

```
import Image from "next-export-optimize-images/image";
        <Image
          src="../../images/profile-photo.jpg"
          width={600}
          height={768}
          className="m-auto mb-16 lg:mb-12"
          priority
          alt="Black and white headshot photo"
        />
```

Removing the image component results in the same failure. Removing both the import and component results in sucess. It is as if next is not acknowledging this isn't the default image component.

Also removed .next, out and node_modules folder and tried a new install and build. Same issue.

**Comment by haydnjonesdesign** [comment]
Issue exists in examples/common dir.

Steps to reproduce:
```
git clone https://github.com/dc7290/next-export-optimize-images.git
```
```
npm i
```
Update package.json to remove next export
```
"generate": "next build && next-export-optimize-images",
```
Update next.config.js to include export
```
const withExportImages = require('next-export-optimize-images')

/**
 * @type {import('next').NextConfig}
 */
const config = {
  output: "export",
  reactStrictMode: true,
  trailingSlash: true,
  eslint: {
    ignoreDuringBuilds: true,
  },
}

module.exports = withExportImages(config)
```
Error
```
> next build && next-export-optimize-images

info - [next-export-optimize-images]: Configuration was not loaded. (This is optional.)
  ▲ Next.js 14.2.5

   Skipping linting
 ✓ Checking validity of types    
   Creating an optimized production build ...
 ✓ Compiled successfully
 ✓ Collecting page data    
 ✓ Generating static pages (3/3)
 ✓ Collecting build traces    

> Image Optimization using the default loader is not compatible with export.
  Possible solutions:
    - Use `next start` to run a server, which includes the Image Optimization API.
    - Configure `images.unoptimized = true` in `next.config.js` to disable the Image Optimization API.
  Read more: https://nextjs.org/docs/messages/export-image-api
   Finalizing page optimization  . ELIFECYCLE  Command failed with exit code 1.
```

**Comment by nfrank** [comment]
# Potential (short term?) fix

## TL;DR
Try setting `"loader": "custom",` in next.config.mjs
like:
```
  images: {
    deviceSizes: [768, 960, 1280, 1600, 1920],
    loader: 'custom',
  },
````

---

## Details, perhaps helpful for a longer term solution.

### 1. The error: 

> Image Optimization using the default loader is not compatible with export.
>   Possible solutions:
>     - Use \`next start\` to run a server, which includes the Image Optimization API.
>     - Configure images.unoptimized = true\ in \`next.config.js\` to disable the Image Optimization API.
>   Read more: https://nextjs.org/docs/messages/export-image-api

is thrown from [next/src/export/index.ts#L285](https://github.com/vercel/next.js/blob/77dac1c8b98cfb6060f865394239b00450cc117f/packages/next/src/export/index.ts#L285) 

All 4 booleans evaluate to true even though `loader` is !== the default nextjs image loader (it's a next-export-optimize-images custom loader).

### 2. Following that back, to the defaults in the code:
```
  const {
    i18n,
    images: { loader = 'default', unoptimized },
  } = nextConfig
```
[next/src/export/index.ts#L264](https://github.com/vercel/next.js/blob/77dac1c8b98cfb6060f865394239b00450cc117f/packages/next/src/export/index.ts#L264)

If in the nextjs config there's no images block and there's no images.loader, the value defaults to `default` which triggers the error to be thrown.

Setting the `images.loader` value to the string `custom` avoids the default loader and passes the validation, then the next-export-optimize-images custom loader picks up (as long as you don't pass in a custom loader file config key:`images.loaderFile` or [specify a custom loader in the props](https://github.com/dc7290/next-export-optimize-images/blob/f590a8e26ee2f60552bc9964f98d664ede1db984/src/components/image.tsx#L14)).

Also of note, the build command I have set to be production: `"build-with-images": "NODE_ENV=production next build && next-export-optimize-images",` so that it doesn't skip the optimizing in the dev/local environment.

(I've been seeing some potential issues with the deviceSizes and other sizes not being respected, but that seems like a different issue, leaving a mention of it here in case you also run into it and consider it related.)

### Example next.config.mjs
(Most of this is just changing some directories, I'm working on some simpler deviceSizes and imageSizes, but not needed for this example)
```
/** @type {import('next').NextConfig} */
import withExportImages from "next-export-optimize-images";

const nextConfig = {
  output: 'export',
  distDir: 'dist',
  reactStrictMode: true,
  images: {
    // deviceSizes: [640, 750, 828, 1200, 1920],
    // imageSizes: [96, 128, 256, 384],
    loader: 'custom'
  },
  trailingSlash: true,
};

export default withExportImages(nextConfig);
```

### Example export-images.config.js
This is all typical config aspects from the documentation, nothing big here
```
const isModeBuild = process.env.NEXT_MODE === 'build'

/**
 * @type {import('next-export-optimize-images').Config}
 */
const config = {
  "mode": 'export',
  "outDir": "dist",
  "convertFormat": [
    ['png', 'webp'],
    ['jpg', 'webp'],
  ],
  "cacheDir": ".image-cache",
  generateFormats: ["webp"]
}
// you may not need the config.mode portion
if(isModeBuild){
  config.mode = 'build'
}

module.exports = config
```

(in my configs I'm also doing some other aspects that you likely wont need, but enclosing for completeness)

@haydnjonesdesign , hope something similar also works for you. The key piece was `loader: 'custom',` Also nice job with the completeness of your writeup above.

**Comment by haydnjonesdesign** [comment]
@nfrank Thanks for an excellent response and investigation to the issue. I'm a bit short on time at the moment to confirm this work-around fixes the issue but will attempt to revisit it in the near future.

Thanks again.

**Comment by dc7290** [comment]
I used to set `loader: 'custom'` on the library side,
However, at some point we decided it was no longer necessary and removed it.
However, now that we know there is such a problem, we'll make sure to set it on the library side again.
Thanks for the report!
